File: src/SeatCanvas/platform/apple/swift/ZoomLevel.swift
//
//  ZoomLevel.swift
//  SeatCanvas
//
//  Created by king on 2026/2/10.
//

import Foundation

@objc(KKSeatCanvasZoomLevel)
@objcMembers
public final class ZoomLevel: NSObject, Sendable {
    /// 最近视角（座位级别）
    public let seat: CGFloat
    /// 行/小块级视角
    public let row: CGFloat
    /// 区域级视角
    @objc(region)
    public let zone: CGFloat
    /// 全场/概览级视角（默认作为彩虹图与座位渲染切换的阈值）
    public let venue: CGFloat

    package init(seat: CGFloat, row: CGFloat, zone: CGFloat, venue: CGFloat) {
        self.seat = seat
        self.row = row
        self.zone = zone
        self.venue = venue
    }

    package static let defalut = ZoomLevel(seat: 1.0, row: 1.0, zone: 1.0, venue: 1.0)
}

File: android/SeatCanvasSample/libseatcanvas/src/main/java/com/libseatcanvas/ZoomLevel.kt
package com.libseatcanvas

/**
 * 缩放级别配置，对应 C++ 层的 ZoomLevelConfig。
 *
 * @property seat 最近视角（座位级别）
 * @property row 行/小块级视角
 * @property zone 区域级视角
 * @property venue 全场/概览级视角
 */
data class ZoomLevel(
    val seat: Float,
    val row: Float,
    val zone: Float,
    val venue: Float
)

